docs/src/design/decoder/constraints.md

```diff
@@ -129,28 +131,28 @@ $$
 
 Using the above variables, we define operation values as described below.
 
-When `JOIN` operation is executed, a new hasher is initialized and contents of $h_0, ..., h_7$ are absorbed into the hasher. We also indicate that the number of elements to be hashed is $8$:
+When `JOIN` operation is executed, a new hasher is initialized and contents of $h_0, ..., h_7$ are absorbed into the hasher. The number of elements to be hashed is $8$ so the $\alpha_4$ term is set to 0:
 
 $$
-u_{join} = f_{join} \cdot (h_{init8} + \alpha_4 \cdot 8) \text{ | degree} = 7
+u_{join} = f_{join} \cdot h_{init8} \text{ | degree} = 7
 $$
 
-When `SPLIT` operation is executed, a new hasher is initialized and contents of $h_0, ..., h_7$ are absorbed into the hasher. We also indicate that the number of elements to be hashed is $8$:
+When `SPLIT` operation is executed, a new hasher is initialized and contents of $h_0, ..., h_7$ are absorbed into the hasher. The number of elements to be hashed is $8$ so the $\alpha_4$ term is set to 0:
 
 $$
-u_{split} = f_{split} \cdot (h_{init8} + \alpha_4 \cdot 8) \text{ | degree} = 7
+u_{split} = f_{split} \cdot h_{init8} \text{ | degree} = 7
 $$
 
-When `LOOP` operation is executed, a new hasher is initialized and contents of $h_0, ..., h_3$ are absorbed into the hasher. We also indicate that the number of elements to be hashed is $4$:
+When `LOOP` operation is executed, a new hasher is initialized and contents of $h_0, ..., h_3$ are absorbed into the hasher. The number of elements to be hashed is $4$ so the $\alpha_4$ term is present:
 
 $$
-u_{loop} = f_{loop} \cdot (h_{init4} + \alpha_4 \cdot 4) \text{ | degree} = 7
+u_{loop} = f_{loop} \cdot (h_{init4} + \alpha_4) \text{ | degree} = 7
 $$
 
-When `SPAN` operation is executed, a new hasher is initialized and contents of $h_0, ..., h_7$ are absorbed into the hasher. We also indicate that the number of operation groups to be hashed is located in the `group_count` register:
+When `SPAN` operation is executed, a new hasher is initialized and contents of $h_0, ..., h_7$ are absorbed into the hasher. The number of operation groups to be hashed is padded to a multiple of the rate width ($8$) and so the $\alpha_4$ is set to 0:
 
 $$
-u_{span} = f_{span} \cdot (h_{init8} + \alpha_4 \cdot gc) \text{ | degree} = 7
+u_{span} = f_{span} \cdot h_{init8} \text{ | degree} = 7
 $$
```

Comment:
We should probably add similar paragraphs for `CALL` and `SYSCALL` operations and integrate $u_{call}$ and $u_{syscall}$ into the formula on lines 172 - 175.

Reply:
Done.